### PR TITLE
feat: add better YAML syntax error messages

### PR DESCRIPTION
**Pull request by char0n**
Refs #2914
Refs #2889

---

Assigning a constant error message is happening as late as possible - during the YAML AST -> ApiDOM syntactic analysis. 

The error message format is consistent with the established error message formats:

1. `(Error <value>)`
2. `(Unexpected <value>)`

Tests in `apidom-ls` have been adapted.

### CODE REVIEW

File: packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts [hunk 1]
```diff
@@ -96,4 +96,34 @@ describe('adapter-browser', function () {
       assert.isTrue(parseResult.isEmpty);
     });
   });
+
+  context('given YAML 1.2 containing a syntax error(1)', function () {
```
Comment (char0n) [inline packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts:100]:
```suggestion
  context('given invalid YAML 1.2 with indentation syntax error', function () {
```

File: packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts [hunk 2]
```diff
@@ -96,4 +96,34 @@ describe('adapter-browser', function () {
...
+    specify('return YAML Syntax error as an annotation in parsing result', async function () {
```
Comment (char0n) [inline packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts:101]:
```suggestion
    specify('should produce syntax error annotation', async function () {
```

File: packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts [hunk 3]
```diff
@@ -96,4 +96,34 @@ describe('adapter-browser', function () {
...
+      const syntaxErrorSpec = `
+        asyncapi: 2.4.0
+        info:
+          version: '1.0.0'
+           title: Something # Badly indented
+      `;
+
```
Comment (char0n) [inline packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts:108]:
```suggestion
```

File: packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts [hunk 4]
```diff
@@ -96,4 +96,34 @@ describe('adapter-browser', function () {
...
+      const parseResult = await adapter.parse(syntaxErrorSpec, { sourceMap: true });
+
+      assert.isTrue(parseResult.errors.toValue()[0] === '(Error YAML syntax error)');
```
Comment (char0n) [inline packages/apidom-parser-adapter-yaml-1-2/test/adapter-browser.ts:111]: Let's try to make this assertion a bit more idiomatic.

---

We need to import `toValue` to translate ApiDOM into JavaScript.
```js
import { toValue } from '@swagger-api/apidom-core';
```

```suggestion
      assert.strictEqual(toValue(parseResult.errors.get(0)), '(Error YAML syntax error)');
```

**Review by char0n — COMMENTED** [review]
@damian-polewski-sb please have a look at my suggestions to tests and if you agree please amend the rest of the tests with the suggestions.

**Comment by damian-polewski-sb** [comment]
@char0n Thanks for the suggestions! I added them in the latest commit.

**Pull request merged**